superbench/benchmarks/micro_benchmarks/cpu_stream_performance.py

```diff
@@ -0,0 +1,134 @@
+
+import os
+
+from superbench.common.utils import logger
+from superbench.benchmarks import BenchmarkRegistry
+from superbench.benchmarks.micro_benchmarks import MicroBenchmarkWithInvoke
+
+
+class CpuStreamBenchmark(MicroBenchmarkWithInvoke):
+    """The Stream benchmark class."""
+
+    def __init__(self, name, parameters=''):
+        """Constructor.
+
+        Args:
+            name (str): benchmark name.
+            parameters (str): benchmark parameters.
+        """
+        super().__init__(name, parameters)
+
+        self._bin_name = 'streamZen3.exe'
+        self.__cpu_arch = ['other', 'zen3', 'zen4']
+
+    def add_parser_arguments(self):
+        """Add the specified arguments."""
+        super().add_parser_arguments()
+
+        self._parser.add_argument(
+            '--CPU_ARCH',
+            type=str,
+            default=['other'],
+            required=False,
+            help='The targeted cpu architectures to run \
+                STREAM. Possible values are {}.'.format(' '.join(self.__cpu_arch))
+        )
+        self._parser.add_argument(
+            '--cores',
+            nargs='+',
+            type=int,
+            required=True,
+            help='List of cores to perform test'
+        )
+
+    def _preprocess(self):
+        """Preprocess/preparation operations before the benchmarking.
+
+        Return:
+            True if _preprocess() succeed.
+        """
+        if not super()._preprocess():
+            return False
+        if not self._set_binary_path():
+            return False
+
+        # zen3
+        # cores=[0, 4, 8, 12, 16, 20, 24, 28, 30, 34, 38, 42, 46, 50,
+        # 54, 58, 60, 64, 68, 72, 76, 80, 84, 88, 90, 94, 98, 102, 106, 110, 114, 118]
+        # zen4
+        # cores=[0, 8, 16, 24, 32, 38, 44, 52, 60, 68, 76, 82, 88, 96, 104, 112, 120,
+        # 126, 132, 140, 148, 156, 164, 170]
+
+        # parse cores argument
+        omp_places = ''
+        for core in self._args.cores:
+            omp_places += '{' + '{}:1'.format(core) + '}'
+
+        envar = 'OMP_SCHEDULE=static && OMP_DYNAMIC=false && OMP_MAX_ACTIVE_LEVELS=1 && OMP_STACKSIZE=256M && \
+            OMP_PROC_BIND=true && OMP_NUM_THREADS={} && OMP_PLACES={}'.format(len(self._args.cores), omp_places)
+
+        if self._args.CPU_ARCH == 'zen3':
+            exe = 'streamZen3.exe'
+        elif self._args.CPU_ARCH == 'zen4':
+            exe = 'streamZen4.exe'
+        else:
+            exe = 'streamx86.exe'
+
+        command = envar + " " + os.path.join(self._args.bin_dir, exe)
+        self._bin_name = exe
+        stream_path = os.path.join(self._args.bin_dir, self._bin_name)
+        ret_val = os.access(stream_path, os.X_OK | os.F_OK)
+        if not ret_val:
+            print("fail3")
+            logger.error(
+                'Executable {} not found in {} or it is not executable'.format(self._bin_name, self._args.bin_dir)
+            )
+            return False
```

Comment:
Call  `if not self._set_binary_path():` after `self._bin_name = exe` directly? There has executable file checking logic in `self._set_binary_path()`

Reply:
addressed